cmd/coder/urls.go

```diff
@@ -14,13 +15,124 @@ import (
 )
 
 type urlsCmd struct{}
+type createURLCmd struct{}
+type delURLCmd struct{}
 
+// DevURL is the parsed json response record for a devURL from cemanager
 type DevURL struct {
+	ID     string `json:"id"`
 	URL    string `json:"url"`
 	Port   string `json:"port"`
 	Access string `json:"access"`
 }
 
+var urlAccessLevel = map[string]string{
+	"PRIVATE": "Only you can access",
+	"ORG":     "All members of your organization can access",
+	"AUTHED":  "Authenticated users can access",
+	"PUBLIC":  "Anyone on the internet can access this link",
+}
+
+func isAccessLevelValid(level string) bool {
+	_, ok := urlAccessLevel[level]
+	return ok
+}
+
+func (cmd createURLCmd) Spec() cli.CommandSpec {
+	return cli.CommandSpec{
+		Name:  "create",
+		Usage: "<env name> <port> [<access>]",
+		Desc: `create/update a devurl for external access
+  <access> is one of [PRIVATE | ORG | AUTHED | PUBLIC]`,
+	}
+}
+
+// Run() creates or updates a devURL, specified by env ID and port
+// (fl.Arg(0) and fl.Arg(1)), with access level (fl.Arg(2)) on
+// the cemanager.
+func (cmd createURLCmd) Run(fl *pflag.FlagSet) {
+	var envName = fl.Arg(0)
+	var port = fl.Arg(1)
+	var access = fl.Arg(2)
+
+	if envName == "" {
+		exitUsage(fl)
+	}
+
+	if access == "" {
+		access = "PRIVATE"
+	}
```

Comment:
Should there also be a check for empty port?

Reply:
I can add, but found the cli package is pretty robust, as well as the APIs we hit. If we don't want to see any of the backend API errors though I can try to validate port more thoroughly here

Reply:
Added port checks